internal/app/components/courseSelections/courseSelections_test.go

```diff
@@ -243,17 +317,146 @@ func TestCourseSelection_IsSatisfied(t *testing.T) {
 			},
 			{
 				CourseRecords: courses.CourseRecords{
-					courses.CourseId(3): &courses.CourseRecord{
+					&courses.CourseRecord{
 						Course: courses.Course{
 							Id: 3,
 						},
 						Grade:          60,
 						CompletionDate: &currTime,
 					},
+					&courses.CourseRecord{
+						Course: courses.Course{
+							Id: 1,
+						},
+						Grade:          59,
+						CompletionDate: &currTime,
+					},
+				},
+			},
+			{
+				CourseRecords: courses.CourseRecords{
+					&courses.CourseRecord{
+						Course: courses.Course{
+							Id: 1,
+						},
+						Grade:          60,
+						CompletionDate: &currTime,
+					},
 				},
 			},
 		},
 		Plans: csEconPlan,
 	}
-	assert.Equal(t, false, courseSelection.IsSatisfied())
+	inCompletePlans = courseSelection.IncompletePlans()
+	assert.Equal(t, 0, len(inCompletePlans))
+}
+
+func TestCourseSelection_InvalidCourses(t *testing.T) {
+	course2 := courses.CourseRecord{
+		Course: courses.Course{
+			Id: 1,
+			Prereqs: courses.CourseRequirement{
+				Course: &courses.Course{
+					Id: 0,
+				},
+				MinGrade: 60,
+			},
+		},
+	}
+	course3 := courses.CourseRecord{
+		Course: courses.Course{
+			Id: 2,
+			Prereqs: courses.CourseRequirement{
+				Course: &courses.Course{
+					Id: 1,
+				},
+				MinGrade: 60,
+			},
+		},
+	}
+	currTime := time.Now()
+	type fields struct {
+		Id             CourseSelectionId
+		Name           CourseSelectionName
+		TermSelections []*TermSelection
+		Plans          plans.Plans
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   courses.CourseRecords
+	}{
+		{
+			name: "all valid",
+			fields: fields{
+				TermSelections: []*TermSelection{
+					{
+						CourseRecords: courses.CourseRecords{
+							&courses.CourseRecord{
+								Course: courses.Course{
+									Id: 0,
+								},
+							},
+						},
+					},
+					{
+						CourseRecords: courses.CourseRecords{
+							&course2,
+						},
+					},
+					{
+						CourseRecords: courses.CourseRecords{
+							&course3,
+						},
+					},
+				},
+			},
+			want: courses.CourseRecords{},
+		},
+		{
+			name: "second and third term invalid",
+			fields: fields{
+				TermSelections: []*TermSelection{
+					{
+						CourseRecords: courses.CourseRecords{
+							&courses.CourseRecord{
+								Course: courses.Course{
+									Id: 0,
+								},
+								Grade:          59,
+								CompletionDate: &currTime,
+							},
+						},
+					},
+					{
+						CourseRecords: courses.CourseRecords{
+							&course2,
+						},
+					},
+					{
+						CourseRecords: courses.CourseRecords{
+							&course3,
+						},
+					},
+				},
+			},
+			want: courses.CourseRecords{
+				&course2,
+				&course3,
+			},
+		},
+	}
+	for _, tt := range tests {
```

Comment:
no idea what's going on here lol but i'll pretend what you're doing is right

Reply:
tests is like an array of test, going through each test and diffing expected vs received. this is part of the test template